python/ray/tests/test_runtime_env_complicated.py

```diff
@@ -46,6 +47,41 @@
 _WIN32 = os.name == "nt"
 
 
+@pytest.fixture(scope="session")
+def conda_env_no_ray():
+    """Creates a conda env without ray installed."""
+    env_name = "conda-no-ray"
+
+    def delete_env(env_name):
```

Comment:
These sub-functions are very similar to `def conda_envs(...)`. Can we add some utility functions to avoid duplicated code?